drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/rightpanel/RightPanelPresenter.java

```diff
@@ -66,22 +79,71 @@ public String getTitle() {
         return ScenarioSimulationEditorConstants.INSTANCE.testTools();
     }
 
-
     @WorkbenchPartView
     public Widget asWidget() {
         return view.asWidget();
     }
 
     @Override
-    public void onEditorTabActivated() {
-        view.showEditorTab();
-        view.hideCheatSheetTab();
+    public void onClearSearch() {
+        view.clearInputSearch();
+        view.hideClearButton();
+    }
+
+    @Override
+    public void onClearNameField() {
+        view.clearNameField();
+    }
+
+    @Override
+    public void onClearStatus() {
+        onClearSearch();
+        onClearNameField();
     }
 
     @Override
-    public void onCheatSheetTabActivated() {
-        view.showCheatSheetTab();
-        view.hideEditorTab();
+    public FactModelTree getFactModelTree(String factName) {
+        return factTypeFieldsMap.get(factName);
+    }
+
+    @Override
+    public void setFactTypeFieldsMap(SortedMap<String, FactModelTree> factTypeFieldsMap) {
+        this.factTypeFieldsMap = factTypeFieldsMap;
+        view.getListContainer().removeAllChildren();
+        this.factTypeFieldsMap.forEach(this::addListGroupItemView);
+    }
+
+    @Override
+    public void onShowClearButton() {
+        view.showClearButton();
+    }
+
+    @Override
+    public void onSearchedEvent(String search) {
+        DivElement listContainer = view.getListContainer();
```

Comment:
You might not want to leak the UI elements into the presenter here. If there are 500+ classes the UI gets so slow that this code will not run properly. I would do the search on the DMO and then fill the UI based on that.

And check with UX what to do if there is thousands of classes in the list. Should we just show the first 20 or what.

Reply:
For now I would just make this work against DMO and make a ticket for the bloated list. Or we can leave it as it is for now and make a ticket for both.

Reply:
@Rikkola good shot, thx!
Fixed with
[DROOLS-2909] Changed search logic. Filtering original fact map and set view's items with filtered entries